src/app/plugins/connectfour.plugin.ts

```diff
@@ -96,112 +100,214 @@ class ConnectFourGame {
 
   private currentPlayer: number = -1;
 
-  private winner: Maybe<number> = null;
-  private tie: boolean = false;
+  private _playingLion: boolean;
+  private _aiDepth = 4;
+  // Difficulty should be given in [1, 100];
+  private _aiDifficulty = 50;
+
+  private _winner: Maybe<number> = null;
+  private _tie: boolean = false;
+  private gameOver: boolean = false;
 
   private _searchDX: number[] = [-1, -1, 0, 1, 1, 1, 0, -1];
   private _searchDY: number[] = [0, -1, -1, -1, 0, 1, 1, 1];
 
-  constructor(playerA: User, playerB: User) {
+  constructor(playerA: User, playerB: User, playingLion: Maybe<boolean>) {
     this._playerA = playerA;
     this._playerB = playerB;
 
+    this._playingLion = playingLion ? true : false;
```

Comment:
silly ternary.

Reply:
Oops, looked like I dumped this change before I pushed.